README.md

```diff
@@ -93,29 +93,65 @@ The MDM Connector code heavily relies on the IDS Components (IAIS git, for acces
+This resource also contains artifacts corresponding to contract offer format for artifact, description and contract containing the permissions and prohibitions. 
+For removing each contract is only deactivated through its corresponding the artifact.
+
+
 ### Directory Watcher
+Directory watcher class takes care that all the artifacts available in this directory of directory watcher are being monitored. Hence it monitors external file IO operations as events which includes permissions to local files. Permissions are taken care with the help of a key provided. Directory Watcher keeps track of active artifact listeners available and notifies in case any discrepancies happening with artifacts while modifying them. Each event is created new or modified or deleted with the help of a watch service.
+A watch service that watches registered objects for changes and events. For example, a file manager may use a watch service to monitor a directory for changes so that it can update its display of the list of files when files are created or deleted.
+
 
 ### Contract Util
+Contract Util Class handles creation of new contract offers and contract agreements via the incoming Contract Requests for the consumers and providers. Incoming contract request are compared with Contract Offer coming from Artifact Listener. Additional functionalities, also include the comparison of contracts.
 
 ### RemoteBrokerLogging
+Provides interactor component class named LoggingInteractor for sending Message and Payload to an External Logging Service, where Inclusion of DAPS might be possible. Another component class to interact with remote broker is to Remotecomponentinteractor. Hence all the remote broker operations like registers and deregistering or updating are handled by Remote Broker Logging class.
 
 ### Handlers
+Handlers are message handlers that process the incoming request messages and send the response as per the required incoming request.
+
+Information as add on:
+ArtifactIndex provides all the required functions for Artifacts, Contracts, Negotiations and makes sure that in case they are present and extends ArtifactFileProvider interface which provides the file. 
+ 
 #### ArtifactWithContractHandler
 
+Handles the incoming request for the artifact provides the requested artifact. Contract information is fetched from the requested artifact. The generation of artifact response is dependent on whether the request has a valid contract or not. If artifact has a contract attached with it, then a Contract Agreement is looked upon further. 
+
 #### ContractAgreementMessageHandler
+Checks whether a ContractAgreement is acceptable or should be rejected. Interacts with Negotiations Service application. (Need more information on Negotiation Service??) . Check if the incoming requested Contract agreement and its sender(consumer) follow the contract offer standards mentioned in ContractUtil (as explained above).
 
```

Comment:
Please remve the reference to the Negotiiation Service

Reply:
Rectified and removed the reference of negotiation service 'Interacts with Negotiations Service application'